### DISCUSSION TITLE
Mobile/desktop breakpoints

### DISCUSSION THREAD

**Discussion by frankhuurman** [zauberzeug/nicegui#4111] [Q&A] [answered]
### Question

I've been trying something simple for way too long and I think I just need a fresh perspective.

I want one button to be visible when on mobile and hidden on all bigger screens and I want the other button to be hidden on mobile and visible on anything bigger than mobile.
I'm sure I had this stuff down with the whole mobile-first tailwind approach and using sm, md and lg but I can't seem to get it done anymore:

```py
from nicegui import ui

ui.button("visible on mobile only").classes("md:hidden")
ui.button("hidden on mobile, visible on laptop and such").classes("hidden md:block")

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @frankhuurman,

Your second button will always have the class "hidden", which is not reliably overwritten by "block".
It's better to use an upper bound for the "hidden" class:

```py
ui.button("mobile").classes("md:hidden")
ui.button("desktop").classes("max-md:hidden")
```

Similar problem: https://github.com/zauberzeug/nicegui/discussions/1333#discussioncomment-6676367

**Reply by frankhuurman** [reply]
I learned about upper bounds today in this manner, thanks Falko :smile:
